src/app/models/propTypes/general/index.js

```diff
@@ -1,12 +1,67 @@
-import { shape, string, oneOf, arrayOf, oneOfType } from 'prop-types';
+import {
+  shape,
+  string,
+  oneOf,
+  arrayOf,
+  oneOfType,
+  checkPropTypes,
+} from 'prop-types';
+
+export const blockObjectOfSpecificTypeAndModel = (type, model) => ({
+  blockId: string.isRequired,
+  type: oneOf([type]).isRequired,
+  model: shape(model).isRequired,
+});
 
 export const blockOfSpecificTypeAndModel = (type, model) =>
-  shape({
-    blockId: string.isRequired,
-    type: oneOf([type]).isRequired,
-    model: shape(model).isRequired,
-  });
+  shape(blockObjectOfSpecificTypeAndModel(type, model));
 
 export const blocksWithTypes = blockTypes => ({
   blocks: arrayOf(oneOfType(blockTypes).isRequired).isRequired,
 });
+
+const requiredPropCheck = (propData, propTypes) => {
+  // get types of required props
+  const requiredProps = propTypes
+    .filter(({ isRequired }) => isRequired)
+    .map(({ type }) => type);
+
+  // get types of passed props
+  const propDataTypes = propData.map(({ type }) => type);
+
+  // get types of required props that are not passed
+  return requiredProps.reduce((missingProps, requiredProp) => {
+    if (!propDataTypes.includes(requiredProp)) {
+      missingProps.push(requiredProp);
+    }
+
+    return missingProps;
+  }, []);
+};
+
+export const arrayOfSpecificBlocks = propTypeData => (props, key) => {
```

Comment:
This method is quite complex - I think it should be pulled out into a separate file to have tests run against it.